src/Workspaces/Core/MSBuild/MSBuild/VisualBasic/VisualBasicProjectFile.cs

```diff
@@ -0,0 +1,468 @@
+            {
+                outputFilePath = this.GetAbsolutePath(outputFilePath);
+            }
+
+            var outputRefFilePath = project.ReadPropertyString("TargetRefPath");
+            if (!string.IsNullOrWhiteSpace(outputRefFilePath))
+            {
+                outputRefFilePath = this.GetAbsolutePath(outputRefFilePath);
+            }
+
+            var targetFramework = project.ReadPropertyString("TargetFramework");
+            if (string.IsNullOrWhiteSpace(targetFramework))
+            {
+                targetFramework = null;
+            }
+
+            var docs = this.GetDocumentsFromModel(project)
+                .Where(s => !IsTemporaryGeneratedFile(s.ItemSpec))
+                .Select(s => MakeDocumentFileInfo(projectDirectory, s))
+                .ToImmutableArray();
+
+            var additionalDocs = this.GetAdditionalFilesFromModel(project)
+                .Select(s => MakeAdditionalDocumentFileInfo(projectDirectory, s))
+                .ToImmutableArray();
+
+            return ProjectFileInfo.Create(
+                this.Language,
+                project.FullPath,
+                outputFilePath,
+                outputRefFilePath,
+                targetFramework,
+                commandLineArgs,
+                docs,
+                additionalDocs,
+                this.GetProjectReferences(project),
+                this.Log);
+        }
+
+        private ImmutableArray<string> FixPlatform(ImmutableArray<string> commandLineArgs)
+        {
+            string platform = null, target = null;
+            var platformIndex = -1;
+
+            for (int i = 0; i < commandLineArgs.Length; i++)
+            {
+                var arg = commandLineArgs[i];
+
+                if (platform == null && arg.StartsWith("/platform:", StringComparison.OrdinalIgnoreCase))
+                {
+                    platform = arg.Substring("/platform:".Length);
+                    platformIndex = i;
+                }
+                else if (target == null && arg.StartsWith("/target:", StringComparison.OrdinalIgnoreCase))
+                {
+                    target = arg.Substring("/target:".Length);
+                }
+
+                if (platform != null && target != null)
+                {
+                    break;
+                }
+            }
+
+            if (string.Equals("anycpu32bitpreferred", platform, StringComparison.OrdinalIgnoreCase)
+                && (string.Equals("library", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("module", target, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals("winmdobj", target, StringComparison.OrdinalIgnoreCase)))
+            {
+                return commandLineArgs.SetItem(platformIndex, "/platform:anycpu");
+            }
+
+            return commandLineArgs;
+        }
+
+        private ImmutableArray<string> ReadCommandLineArguments(MSB.Execution.ProjectInstance project)
```

Comment:
>ReadCommandLineArguments [](start = 39, length = 24)

This code looks very similar to what we have to do in the compiler in our MSBuild task code. Wonder if there is a way to combine this. If not I need to add updating this to our "TODO" list whenever we add a new switch.

Reply:
This is just a best effort to get a handful of properties. It should only happen in the case where the build failed to produce command-line arguments. So, I'm not sure it makes sense to merge with the MSBuild task.

Reply:
Looking a bit more deeply, I still don't think it's a good idea. This particular bit of code runs when the build didn't return any CscCommandLineArgs or VbcCommandLineArgs. That means that the build failed in some way and the MSBuild ProjectInstance is likely in an incomplete state. At that point, it's really in recovery mode -- just trying to get *something* back to the user. I don't know that we want that sort of logic in the MSBuild task.